File: 0001_Two_Sum/0001_Two_Sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var targetValues:[Int: Int] = [:]

        for i: Int in 0...nums.count-1 {
            // Found Match
            if let savedValueToAddIndex: Int = targetValues[target - nums[i]] {
                return [i, savedValueToAddIndex]
            }
            // Save index and value
            targetValues[nums[i]] = i
        }

        return [-1, -1]
    }
}

File: 0001_Two_Sum/0001_Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        '''
        targetValues: Store elem's index and value that adds up to the target as key
        - Key: (Target - elem) in nums
        - Value: index of the elem
        '''
        targetValues = {}

        for i in range(len(nums)):
            # Found Match
            if nums[i] in targetValues:
                return [targetValues[nums[i]], i]
            
            # Save Value and Index
            targetValues.update({target - nums[i]: i})
            
        # Not Found
        return [-1, -1]